### PR TITLE
Decorate a class

### PR DESCRIPTION
**Pull request by Mahamadou-Nouridine**
In this Pr, I added changes that include:

#### Interface
- Create a class `Nameable`.
   - Implement a method called `correct_name` that will raise a `NotImplementedError`.

#### Turn Person class to Nameable
- Make sure that Person class inherits from Nameable
- Make sure that this class has a method `correct_name` implemented. It simply returns the name attribute.

#### Prepare base Decorator
- Make sure that it inherits from Nameable.
- In the constructor assign a nameable object from params to an instance variable.
- Implement the `correct_name` method that returns the result of the correct_name method of the `@nameable`.

#### Prepare CapitalizeDecorator and TrimmerDecorator
- For the CapitalizeDecorator:
    - Create a class that inherits from the base Decorator class.
    - Implement a method `correct_name` that capitalizes the output of `@nameable.correct_name`.
- For the TrimmerDecorator:
    - Create a class that inherits from the base Decorator class.
    - Implement a method `correct_name` that ensures that the output of `@nameable.correct_name` has a maximum of 10 characters. If it's longer it should trim the word.

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -1,14 +1,21 @@
-class Person
+require_relative 'nameable'
+
+class Person < Nameable
   attr_accessor :name, :age
   attr_reader :id
 
   def initialize(age, name = 'unknown', parent_permission: true)
+    super
```
Comment (julie-ify) [inline person.rb]: - [x] This should be `super()`, not just `super`, else it returns errors.

File: base_decorator.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
+require_relative 'nameable'
+
+class BaseDecorator < Nameable
+  def initialize(nameable)
+    super
```
Comment (julie-ify) [inline base_decorator.rb]: - [x] This should be `super()`, not just `super`

File: trimmer_decorator.rb [hunk 3]
```diff
@@ -0,0 +1,11 @@
+require_relative 'base_decorator'
+
+class TrimmerDecorator < BaseDecorator
+  def correct_name
+    if super.length <= 10
+      super
+    else
+      super[0..3]
```
Comment (julie-ify) [inline trimmer_decorator.rb]: - [x] We should return 10 characters, not 4. Kindly fix that.

  ```
  super[0..9]
  ```

**Review by julie-ify — CHANGES_REQUESTED** [review]
## Hi @Mahamadou-Nouridine,

Good job so far!

## Highlights!
- Good feature branch name ✔️ 
- No linter errors ✔️ 
- Descriptive pull request ✔️ 
- Descriptive commit messages ✔️ 
- Good code organization and working app ✔️
- Descriptive README ✔️ 

Hello ✋, you have done well reaching this point of the milestone 🤝 . But there are some issues that you still need to work on to go to the next project but you are almost there!

# Status: Required Changes ♻️⛔️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread and tag me `@julie-ify` if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: trimmer_decorator.rb (hunk 3 above)
Comment (Mahamadou-Nouridine) [reply trimmer_decorator.rb]: Thank you

**Review by joel-kalema — APPROVED** [review]
Hi @Mahamadou-Nouridine 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

# Congratulations! 🎉 


## Highlights

The project is running successfully 👍 
PR is well descriptive 👍 
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me **@joel-kalema** in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**